src/v2/models/AppState.js

```diff
@@ -106,6 +106,41 @@ export default Model.extend({
     return currentViewState;
   },
 
+  /**
+   * Returns the displayName of the authenticator
+   * @returns {string}
+   */
+  getAuthenticatorDisplayName () {
+    const isChallenge = this.isAuthenticatorChallenge();
+    const currentAuthenticator = this.get('currentAuthenticator');
+    if (!isChallenge) {
+      // Always returned during enrollment
+      return currentAuthenticator.displayName;
+    }
+
+    // Challenge/verification flows may or may have this name in different
+    // parts of the response. We walk through those here.
+    const currentAuthenticatorEnrollment = this.get('currentAuthenticatorEnrollment');
+    if (currentAuthenticatorEnrollment && currentAuthenticatorEnrollment.displayName) {
+      // Always returned for challenge/verify flows - except when the authenticator is
+      // verified by type.
+      return currentAuthenticatorEnrollment.displayName;
```

Comment:
In [some cases](https://github.com/okta/okta-signin-widget/blob/master/src/v2/view-builder/views/custom-otp/ChallengeCustomOTPAuthenticatorView.js#L11) we fallback to a default vendor name if urrentAuthenticatorEnrollment.displayName is not set.

Reply:
Don't need to handle that in this PR but when we do this in the future, we should think about accepting the authenticator type as an argument to this method.

Reply:
Yeah - from what I understand from the few backend devs I've talked to, the `displayName` is a required property that'll always be returned via our API. Doesn't hurt to have this fallback though!